Module not found: Can't resolve './_next/static/media/b1db3e28af9ef94a-s.woff2'  After set   assetPrefix to '.',

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
❯ pnpm next info

    Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 19.6.0: Sun Nov 14 19:58:51 PST 2021; root:xnu-6153.141.50~1/RELEASE_X86_64
    Binaries:
      Node: 16.18.1
      npm: 8.19.2
      Yarn: 1.22.19
      pnpm: 7.17.1
    Relevant packages:
      next: 13.2.4
      eslint-config-next: 13.2.4
      react: 18.2.0
      react-dom: 18.2.0

warn  - Latest canary version not detected, detected: "13.2.4", newest: "13.2.5-canary.20".
        Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
        Read more - https://nextjs.org/docs/messages/opening-an-issue
```


### Which area(s) of Next.js are affected? (leave empty if unsure)

_No response_

### Link to the code that reproduces this issue

no

### To Reproduce

pnpm create next-app  (with default options) 

modify next.config.js as following

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  assetPrefix: '.',    // or './'
}

module.exports = nextConfig
```

pnpm build

then will get following error
```shell
pnpm build

> next-test@0.1.0 build /Users/lj/Study/web3.0/next-test
> next build

info  - Linting and checking validity of types  
Failed to compile.

src/pages/index.tsx
Module not found: Can't resolve './_next/static/media/d1d9458b69004127-s.woff2'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/.pnpm/next@13.2.4_biqbaboplfbrettd7655fr4n2y/node_modules/next/font/google/target.css?{"path":"src/pages/index.tsx","import":"Inter","arguments":[{"subsets":["latin"]}],"variableName":"inter"}
./src/pages/index.tsx

src/pages/index.tsx
Module not found: Can't resolve './_next/static/media/b967158bc7d7a9fb-s.woff2'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/.pnpm/next@13.2.4_biqbaboplfbrettd7655fr4n2y/node_modules/next/font/google/target.css?{"path":"src/pages/index.tsx","import":"Inter","arguments":[{"subsets":["latin"]}],"variableName":"inter"}
./src/pages/index.tsx

src/pages/index.tsx
Module not found: Can't resolve './_next/static/media/ae9ae6716d4f8bf8-s.woff2'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/.pnpm/next@13.2.4_biqbaboplfbrettd7655fr4n2y/node_modules/next/font/google/target.css?{"path":"src/pages/index.tsx","import":"Inter","arguments":[{"subsets":["latin"]}],"variableName":"inter"}
./src/pages/index.tsx

src/pages/index.tsx
Module not found: Can't resolve './_next/static/media/c0f5ec5bbf5913b7-s.woff2'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/.pnpm/next@13.2.4_biqbaboplfbrettd7655fr4n2y/node_modules/next/font/google/target.css?{"path":"src/pages/index.tsx","import":"Inter","arguments":[{"subsets":["latin"]}],"variableName":"inter"}
./src/pages/index.tsx

src/pages/index.tsx
Module not found: Can't resolve './_next/static/media/b1db3e28af9ef94a-s.woff2'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/.pnpm/next@13.2.4_biqbaboplfbrettd7655fr4n2y/node_modules/next/font/google/target.css?{"path":"src/pages/index.tsx","import":"Inter","arguments":[{"subsets":["latin"]}],"variableName":"inter"}
./src/pages/index.tsx


> Build failed because of webpack errors
info  - Creating an optimized production build . ELIFECYCLE  Command failed with exit code 1.
```


### Describe the Bug

I am not sure what happend, please help

### Expected Behavior

expected `pnpm build` can work ok with assetPrefix config

### Which browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

--- a/packages/next/src/build/webpack/loaders/next-font-loader/index.ts
+++ b/packages/next/src/build/webpack/loaders/next-font-loader/index.ts
@@ -49,6 +49,15 @@ export default async function nextFontLoader(this: any) {
       postcss: getPostcss,
     } = this.getOptions()
 
+    if (assetPrefix && !/^\/|https?:\/\//.test(assetPrefix)) {
+      const err = new Error(
+        'assetPrefix must start with a leading slash or be an absolute URL(http:// or https://)'
+      )
+      err.name = 'NextFontError'
+      callback(err)
+      return
+    }
+
     /**
      * Emit font files to .next/static/media as [hash].[ext].
      *

--- a/test/integration/font-optimization/fixtures/invalid-assertprefix/next.config.js
+++ b/test/integration/font-optimization/fixtures/invalid-assertprefix/next.config.js
@@ -1,0 +1,3 @@
+module.exports = {
+  assetPrefix: '.',
+}

--- a/test/integration/font-optimization/fixtures/invalid-assertprefix/pages/index.js
+++ b/test/integration/font-optimization/fixtures/invalid-assertprefix/pages/index.js
@@ -1,0 +1,10 @@
+import React from 'react'
+import { Inter } from 'next/font/google'
+
+const inter = Inter({ subsets: ['latin'] })
+
+const Page = () => {
+  return <div className={inter.className}>Hello</div>
+}
+
+export default Page

--- a/test/integration/font-optimization/test/index.test.js
+++ b/test/integration/font-optimization/test/index.test.js
@@ -314,7 +314,7 @@ describe('Font Optimization', () => {
     })
   })
   describe('invalid configuration', () => {
-    it('should an error if assetPrefix starts with .', async () => {
+    it('should show a proper error if assetPrefix starts with .', async () => {
       const appDir = join(fixturesDir, 'invalid-assertprefix')
       const { stderr } = await nextBuild(appDir, undefined, {
         stderr: true,

--- a/test/integration/font-optimization/test/index.test.js
+++ b/test/integration/font-optimization/test/index.test.js
@@ -313,4 +313,15 @@ describe('Font Optimization', () => {
       )
     })
   })
+  describe('invalid configuration', () => {
+    it('should an error if assetPrefix starts with .', async () => {
+      const appDir = join(fixturesDir, 'invalid-assertprefix')
+      const { stderr } = await nextBuild(appDir, undefined, {
+        stderr: true,
+      })
+      expect(stderr).toContain(
+        'assetPrefix must start with a leading slash or be an absolute URL(http:// or https://)'
+      )
+    })
+  })
 })

